contrib/src/main/java/com/datatorrent/contrib/geode/GeodeStore.java

```
@@ -0,0 +1,298 @@
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+
+    region = clientCache.getRegion(getRegionName());
+
+    if (region == null) {
+      region = clientCache.<Object, Object> createClientRegionFactory(ClientRegionShortcut.PROXY).create(
+          getRegionName());
+    }
+
+  }
+
+  @Override
+  public void disconnect() throws IOException
+  {
+    clientCache.close();
+
+  }
+
+  @Override
+  public boolean isConnected()
+  {
+    return (clientCache.isClosed());
+
+  }
+
+  /**
+   * Gets the value given the key. Note that it does NOT work with hash values
+   * or list values
+   *
+   * @param key
+   * @return The value.
+   */
+  @Override
+  public Object get(Object key)
+  {
+
+    try {
+      return (getRegion().get(key));
+    } catch (IOException ex) {
+      throw new RuntimeException("Exception while getting the object", ex);      
+
+    }
+
+  }
+
+  /**
+   * Gets all the values given the keys. Note that it does NOT work with hash
+   * values or list values
+   *
+   * @param keys
+   * @return All values for the given keys.
+   */
+  @SuppressWarnings("unchecked")
+  @Override
+  public List<Object> getAll(List<Object> keys)
+  {
+
+    List<Object> values = new ArrayList<Object>();
+
+    try {
+      final Map<Object, Object> entries = getRegion().getAll(keys);
+      for (int i = 0; i < keys.size(); i++) {
+        values.add(entries.get(keys.get(i)));
+      }
+    } catch (IOException ex) {
+      logger.info("error getting region ", ex);
+    }
+
+    return (values);
+
+  }
+
+  /**
+   * @param regionName
+   *          the regionName to set
+   */
+  public void setRegionName(String regionName)
+  {
+    this.regionName = regionName;
+  }
+
+
+  public Map<Object, Object> getAllMap(List<Object> keys)
+  {
+
+    try {
+      final Map<Object, Object> entries = getRegion().getAll(keys);
+      return (entries);
+    } catch (IOException ex) {
+      logger.info("error getting object ", ex);
```

why swallowing exception?